### DISCUSSION TITLE
WalletConnect and Chrome extension

### DISCUSSION THREAD

**Discussion by WiwiC** [WalletConnect/walletconnect-monorepo#1370] [Developer Support] [answered]
Hi,

Does wallet connect works also with a chrome extension ? If yes, does this chrome extension can be listed in the WalletConnect catalogue through the normal process ? Any special requirement to take into consideration ?

Also, let's assume that this wallet extension is communicating with a 'smart contract wallet', could WalletConnect expose the smart contract wallet address to interact with a Dapp but transfer signature and transactions to the user public address owning the private key on behalf of the smart contract wallet ?

Thanks.

**Comment by ganchoradkov** [comment, marked as answer]
Hey @WiwiC,

Walletconnect does work in a chrome extension. It makes things a bit trickier but it's possible and it's been done before.
What you need to do is :

- initialize Walletconnect in the service worker of the extension
- set up a inpage script to be injected as the page loads such as https://www.npmjs.com/package/@metamask/providers
- pump the JSON RPC requests to your background.js
- handle the incoming JSON RPC requests coming from the Dapp & forward them to the walletconnect client

Your flow will be `Dapp <-> Extension <-> Wallet`

Best,
Gancho

**Reply by WiwiC** [reply]
Hi @ganchoradkov, 

Thank you for your response. It helps !
And regarding the second question, what is your view ? I know that in the case of ArgentHQ wallet (integrated with WalletConnect: https://docs.argent.xyz/wallet-connect-and-argent) for instance, WalletConnect allows to interact with a Dapp by exposing directly the smart contract wallet address to the Dapp **although** signatures of transactions are made through the mobile app where the private key of the account (EOA) owning the smart contract wallet is stored. 

To me, it looks like pretty much the same process I want to achieve as I explained above. It does have a slightly different logic though, as I only want to allow some calls & have only partial ownership of the smart contract wallet for the user (i.e. the user is only allowed to do some things with the smart contract wallet).

thanks,
Best,

**Reply by gilles437** [reply]
is there a github repo showing the code of a chrome extension that connects to a wallet using walletConnect?

**Reply by gilles437** [reply]
@WiwiC is there some public version of this code?

**Reply by dhadx** [reply]
> is there a github repo showing the code of a chrome extension that connects to a wallet using walletConnect?

Have you been able to use the walletconnect ??
